mri/operators/linear/wavelet.py

```diff
@@ -57,60 +67,89 @@ def get_coeff(self):
     def set_coeff(self, coeffs):
         self.transform.analysis_data = coeffs
 
+    def _op(self, data):
+        if isinstance(data, np.ndarray):
+            data = pysap.Image(data=data)
+        self.transform.data = data
+        self.transform.analysis()
+        coeffs, coeffs_shape = flatten(self.transform.analysis_data)
```

Comment:
Maybe this question is not for now but for the refactoring in general: do we really need to flatten the coefficients?

Reply:
well legacy code, and I would say it would ensure consistency in the way data is organized and also save a bit on memory (latter I am not very sure though)